crates/bevy_ecs/src/removal_detection.rs

```diff
@@ -139,7 +147,7 @@ impl<'a, 'w, 's: 'a, T> IntoIterator for &'a mut RemovedComponents<'w, 's, T>
 where
     T: Component,
 {
-    type Item = Entity;
+    type Item = RemovedComponentEntity;
```

Comment:
This is a bit unfortunate, ideally the iter would return an `Entity`, but I'm unsure @alice-i-cecile thoughts?

Reply:
I think we can probably deref in this method. I would very much like the item type here to be `Entity` for both ergonomics and to reduce breaking changes.